Add max_align_t typedef for OS versions that don't have it in stddef.h.

--- src/libjasper/include/jasper/jas_types.h.orig	2022-07-14 01:14:38.000000000 +1000
+++ src/libjasper/include/jasper/jas_types.h	2022-09-11 04:02:48.000000000 +1000
@@ -183,6 +183,8 @@ this behavior.
 #	define max_align_t long double
 #elif !(defined(__STDC_VERSION__) && (__STDC_VERSION__ - 0 >= 201112L))
 #	define max_align_t long double
+#elif defined(__apple_build_version__)
+    typedef long double max_align_t;
 #endif
 
 #if 0
